docs/bin/dump_config.py

```diff
@@ -62,10 +63,9 @@ def main(args):
     output_name = os.path.join(output_dir, template_file.replace('.j2', ''))
     temp_vars = {'config_options': config_options}
 
-    with open(output_name, 'wb') as f:
-        f.write(template.render(temp_vars).encode('utf-8'))
+    data = template.render(temp_vars).encode('utf-8')
 
-    return 0
```

Comment:
Please keep 0 return code. This is a way of communicating success to an OS

Reply:
Which is the default, so it's useless.

Reply:
Besides, return here is what the `main()` function returns, not the program's exit code (which would be using `sys.exit(0)`)

Reply:
The place where the `main()` function is called explicitly does `sys.exit(main(sys.argv[:]))`. So `return 0` should really be here.

(And yes, returning nothing is equivalent to `return None`, and `sys.exit(None)` is equivalent to `sys.exit(0)`, but it's still nicer to have an explicit `return 0` IMO.)

Reply:
In that case, you should also remove the `sys.exit()` call around `main(sys.argv[:])` (in `docs/bin/plugin_formatter.py`, there's also no `sys.exit()` around `main()`).

Reply:
I did that when you made me aware of it :-)